src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -271,21 +272,44 @@ public static SyntaxList<AttributeListSyntax> GetAttributeLists(this SyntaxNode
                 _ => default,
             };
 
-        public static ConditionalAccessExpressionSyntax? GetParentConditionalAccessExpression(this SyntaxNode? node)
+        /// <summary>
+        /// <inheritdoc cref="ISyntaxFacts.GetRootConditionalAccessExpression(SyntaxNode)"/>
+        /// </summary>>
+        public static ConditionalAccessExpressionSyntax? GetRootConditionalAccessExpression(this SyntaxNode? node)
```

Comment:
these functions were rewritten for clarity and to match (with docs) their corresponding parsing impls on the compiler side.  Those codepaths were audited to make sure we were handling all cases, and tests were added to make sure we do things properly for all of them.

Reply:
i also renamed this to be clearer that you're not necessarily getting a direct parent, but the root of the CAE tree as i felt the previous name was a bit confusing.